src/legacy/core_plugins/kibana/public/management/sections/objects/lib/resolve_saved_objects.ts

```diff
@@ -135,14 +153,14 @@ function groupByType(docs) {
   }, defaultDocTypes);
 }
 
-async function awaitEachItemInParallel(list, op) {
+async function awaitEachItemInParallel<T, R>(list: T[], op: (item: T) => R) {
   return await Promise.all(list.map(item => op(item)));
 }
 
 export async function resolveIndexPatternConflicts(
-  resolutions,
-  conflictedIndexPatterns,
-  overwriteAll
+  resolutions: Array<{ oldId: string; newId: string }>,
+  conflictedIndexPatterns: any[],
```

Comment:
working with code in this file is not fun 😓

Reply:
`lib/resolve_saved_objects.js` and `lib/resolve_import_errors.js` really could use some proper cleanup and refactoring (most of the lib function does). I will create an issue to track all possible improvements once most of the SO management PR lands.